Finishes #61071 "Combat Adaptation -> Berserk (rework)"

CwasCard opened the pull request:
#### Summary
Features "Reworks Beast's Combat Adaptation into Berserk, adds Berserk fighting style accessible while BERSERK! is active."

#### Purpose of change

Finishes #61071 with a couple of my own additions. 

#### Describe the solution

Combat Adaptation is replaced with Berserk, a mutation that can be activated to turn into BERSERK!. When BERSERK! is active, the character cannot block or dodge, their strength is increased by 14, their intelligence is decreased by 20, they move 30% faster and they cannot feel any new pain. They also gain access to the BERSERK! fighting style which scales heavily off of their now-increased strength. Anywhere from 2 to 5 minutes from activation, the BERSERK! effect wears off, and they instead experience a -6 modifier to their Strength, Perception, and Dexterity, a surge of pain and -30 speed for 10 minutes. They are additionally dazed for 30 seconds.

The BERSERK! style is much stronger while the user has the Natural Stance effect - the damage is higher and the techniques - aside from Overwhelm -  don't require the user to have crit to take effect.

#### Describe alternatives you've considered

Instead of allowing Berserk to be activated manually, change it to be a random activation any time the user gets hit while a hostile is visible. This would require both the buffs and nerfs to be flattened down. I would like to keep the Berserk style in some form, though.

#### Testing

Start a world, give myself Beast threshold, Berserk mutation. Activate, watch effects. Wait for 5 minutes, Berserk comedown starts at 2 minutes. Berserk comedown lasts for 10. Give myself Digitigrade & Paws, activate Berserk again after the comedown, I only use the Natural Stance version of BERSERK!

#### Additional context

This is a question for anyone who's interested in answering - do you have any ideas for how to simulate multiple limbs _without_ the WIP Limb Framework? I'd like to re implement #78982 in accordance with Maleclypse's suggestions (and in so doing give Chimera lots of extra goodies). I have an idea of where to start, but I'm not sure how to meaningfully and satisfyingly implement the "multiple limbs" part. 


also I finally remembered to lint this time haha

CwasCard committed:
Adds Berserk and BERSERK! mutation, adds Berserk fighting style & BERSERK! comedown effect.

CwasCard committed:
Nerfs berserk style, fixes usage of "vector_hand"

CwasCard committed:
Update mutations.json

kevingranade reviewed (changes requested):
These numbers are completely out of line, +4 strength would be high enough that I'd ask for a more in depth analysis and rationale,  +10% speed likewise, and that's a far as I got because your numbers are multiples of that.

CwasCard commented:
> These numbers are completely out of line, +4 strength would be high enough that I'd ask for a more in depth analysis and rationale, +10% speed likewise, and that's a far as I got because your numbers are multiples of that.

Alright. Are +2 and 5% respectively more reasonable?

CwasCard committed:
Nerfs bonuses from BERSERK!, tones down maluses by about equal measure.

Squishes BERSERK's strength & speed down to 2 & 10% respectively. Removes a lot of the damage scaling from BERSERK!'s techniques as well.

CwasCard committed:
Update data/json/martialarts_fictional.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

worm-girl commented:
> > These numbers are completely out of line, +4 strength would be high enough that I'd ask for a more in depth analysis and rationale, +10% speed likewise, and that's a far as I got because your numbers are multiples of that.
> 
> Alright. Are +2 and 5% (and other such squishdowns) more reasonable?

I honestly think just disabling (or seriously reducing) pain maluses for the duration would be good enough to make players want to push the button. If you really wanted to, a short term morale boost from the flood of endorphines or whatever (plus a crash after) wouldn't go amiss. Beast is already quite strong.

GuardianDll commented:
Need to resolve test errors and outdated syntax
